language-server/src/dotty/tools/languageserver/DottyLanguageServer.scala

```diff
@@ -466,12 +471,14 @@ class DottyLanguageServer extends LanguageServer
 
   override def symbol(params: WorkspaceSymbolParams) = computeAsync { cancelToken =>
     val query = params.getQuery
+    def predicate(implicit ctx: Context): NameTree => Boolean =
+      tree => tree.symbol.exists && !tree.symbol.isLocal && tree.name.toString.contains(query)
```

Comment:
Shouldn't symbol and documentSymbol have similar predicates ? I.e., should we also exclude primary constructors and synthetic symbols here ?

Reply:
Yes, that makes a lot of sense. I've removed checks that were redundant with those happening in `namedTrees`.

I'm also thinking that `namedTrees` should be the one excluding primary constructors. I can't think of situations where they should be included, but I may very well be missing something. What do you think?